backend/src/modules/teachers/controller.js

```diff
@@ -0,0 +1,155 @@
+import { Op } from 'sequelize'
+
+import Teachers from '../../models/teachers'
+import Users from '../../models/users'
+import Classes from '../../models/classes'
+import Students from '../../models/students'
+
+import config from '../../configs/teachers'
+
+export default class TeachersController {
+  static async create(teacher) {
+    const { name, userID } = teacher
+
+    const errors = []
+
+    const test = await Users.findOne({
+      where: {
+        id: userID,
+      },
+      include: [{
+        model: Teachers,
+        as: 'teacher',
+      },
+      {
+        model: Students,
+        as: 'student',
+      }],
+    })
+
+    if (!test) {
+      errors.push({
+        msg: 'The user doesn`t exist',
+        param: 'userID',
+        location: 'body',
+      })
+
+      return { errors }
+    }
+
+    if (test.teacher || test.student) {
+      errors.push({
+        msg: 'The user is booked',
+        param: 'userID',
+        location: 'body',
+      })
+
+      return { errors }
+    }
+
+    const teacheExists = await Teachers.findOne({
+      attributes: ['id'],
+      where: {
+        name,
+      },
+    })
+
+    if (teacheExists) {
+      errors.push({
+        msg: 'The teacher exists',
+        param: 'name',
+        location: 'body',
+      })
+
+      return { errors }
+    }
+
+    await Teachers.create(teacher)
+
+    return {
+      created: !!teacher,
+      teacher: teacher || null,
+    }
```

Comment:
You should check whether the Teacher has been created by setting the value of "await Teachers.create(teacher)" to some variable.